### PR TITLE
Add more detailed PR guides in CONTRIBUTING.md

### LINKED ISSUE #2427: Document squash&push in contributing guidelines
### What should be done?

Currently, we say this in PROCESS.md:

https://github.com/FerretDB/FerretDB/blob/5adf466b7fefa0b709370356ea4db9bb882b3a51/.github/PROCESS.md?plain=1#L87-L90

But at least that bit is important for all contributors because squashes and force-pushes make reviews harder for us. I think we should move that information (and maybe something else that is important for everyone) to CONTRIBUTING.md.

### PR DESCRIPTION
**Pull request by AuruTus**
# Description

Closes #2427 .

Add squash&push guides into `CONTRIBUTING.md` for better community involvement. And I also add other guides related with PR management which, I think, are also important.

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [ ] I made spot refactorings.
* [x] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [ ] I ensured that PR title is good enough for the changelog.
* [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by AuruTus** [comment]
Is it a bit long for contributing guides? I only pick some important rules for new contributors, because it seems they cannot access all repo operations. BTW, can reply on next Monday, sorry for my late PR.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -227,7 +227,7 @@ Also, we should use driver methods as much as possible instead of testing comman
 
 ### Submitting code changes
 
-Before submitting a pull request, please make sure that:
+**Before submitting a pull request, please make sure that:**
```
Comment (AlekSi) [inline CONTRIBUTING.md:230]: Let's use subheaders instead:

```markdown
### Before submitting PR

Before submitting a pull request, please make sure that:
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -239,6 +239,22 @@ Before submitting a pull request, please make sure that:
 3. Comments are rendered correctly in the `task godocs` output.
 4. `task all` passes.
 
+**For creating and submitting a pull request, please follow these guides:**
+
+1. Please create a **draft pull request** when you start working on an issue.
+   Once it’s ready for review, mark it as such.
+   You don’t need to switch between draft and non-draft states after that
```
Comment (AlekSi) [inline CONTRIBUTING.md:246]: I think that point could be either removed or reformulated. We created draft PRs mostly to track our work. External contributors may not be comfortable with creating "unready" PRs.

**Review by AlekSi — COMMENTED** [review]
Good work, but I think we should _move_ those points rather than _copy_ them to avoid having two separate copies to maintain.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -239,6 +239,22 @@ Before submitting a pull request, please make sure that:
 3. Comments are rendered correctly in the `task godocs` output.
 4. `task all` passes.
```
Comment (AlekSi) [inline CONTRIBUTING.md:241]:
```markdown
### Submitting PR
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (AuruTus) [reply CONTRIBUTING.md]: This is better. Done it.

File: CONTRIBUTING.md (hunk 2 above)
Comment (AuruTus) [reply CONTRIBUTING.md]: Ok, removed.

**Comment by AuruTus** [comment]
> Good work, but I think we should _move_ those points rather than _copy_ them to avoid having two separate copies to maintain.

OK I removed the redundant points in `PROCESS.md`. I've thought to keep those as a complete working flow.

File: .github/PROCESS.md [hunk 4]
```diff
@@ -64,35 +64,22 @@ Try to estimate on the planning as best as you can.
...
-7. In the pull request review conversations, please either leave a new comment or resolve (close) the conversation,
-   but do not do that simultaneously.
-   That ensures that the other person can read all comments.
-8. Please don't forget to click "re-request review" buttons once PR is ready for re-review.
-9. Commit messages (both titles and bodies) are not important and can be "WIP" or anything else.
-   All commits are always squashed on merge by GitHub, so there is no need to squash them manually, amend them, and/or do force pushes.
-   Autogenerated GitHub's squash commit's body should be manually replaced by "Closes #{issue_number}.".
-   Do not leave original commit messages in the squashed commit body.
-10. Pull requests should be merged by an auto-merge that the author should enable.
+6. Please don't forget to click "re-request review" buttons once PR is ready for re-review.
```
Comment (AlekSi) [inline .github/PROCESS.md:77]: Please move that point to the CONTRIBUTING.md too. And then do it once PR is ready for re-review :)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -239,6 +241,18 @@ Before submitting a pull request, please make sure that:
...
+
+1. If the pull request is related to some issues, please mention the issue number in the pull request **description** like `Closes #{issue_number}.` or `Closes org/repo#{issue_number}.`(You can just follow the pull request template).
+   If you propose a tiny fix, there is no needed to create a new issue.
+2. In the pull request review conversations, please either leave a new comment or resolve (close) the conversation, which ensures the other people can read all comments.
+   But **do not** do that simultaneously.
+3. During your development, commit messages (both titles and bodies) are not important and can be "WIP" or anything else.
+   All commits are always squashed on merge by GitHub, so there is **no need** to squash them manually, amend them, and/or do force pushes.
+   But notice that the autogenerated GitHub's squash commit's body **should be** manually replaced by "Closes #{issue_number}.".
+
+For more details about long-term contributions, please read [PROCESS.md](.github/PROCESS.md)
```
Comment (AlekSi) [inline CONTRIBUTING.md:254]: It is more about long-term contributors (people), not contributions. I think we should say that.

File: .github/PROCESS.md [hunk 6]
```diff
@@ -64,35 +64,22 @@ Try to estimate on the planning as best as you can.
...
+7. Pull requests should be merged by an auto-merge that the author should enable.
     If the pull request is not ready to be merged for some reason (but ready to be reviewed as usual),
     the description should explain why, and the `do not merge` label should be applied.
     Even in that case, auto-merge should still be enabled by the author.
-11. Ultimately, the author is responsible for doing everything to ensure that the pull request is merged and done.
+8. Ultimately, the author is responsible for doing everything to ensure that the pull request is merged and done.
```
Comment (AlekSi) [inline .github/PROCESS.md:83]: I think we should link back to CONTRIBUTING.md somewhere. Maybe there. Maybe before that list. Maybe on top of the file.

File: .github/PROCESS.md (hunk 4 above)
Comment (AuruTus) [reply .github/PROCESS.md:77]: Oh I mixed it with draft things, will move it back to CONTRIBUTING.md

**Comment by AlekSi** [comment]
Could you please click a button/link in that comment: https://github.com/FerretDB/FerretDB/pull/2435#issuecomment-1509445990 ?

**Comment by AuruTus** [comment]
> Could you please click a button/link in that comment: [#2435 (comment)](https://github.com/FerretDB/FerretDB/pull/2435#issuecomment-1509445990) ?

OK done with it. I even didn't notice that button before 😅.

**Review by AlekSi — APPROVED** [review]

**Comment by AlekSi** [comment]
Thank you!

**Pull request merged**